app/jobs/webhook_deliveries/deliver_job.rb

```diff
@@ -0,0 +1,40 @@
+module WebhookDeliveries
+  class DeliverJob < ApplicationJob
+    queue_as :default
+
+    def perform(event_type, course, resource)
+      webhook_endpoint = course.webhook_endpoint
+      payload = {
+        data: data(event_type, resource),
+        event: event_type
+      }
+      uri = URI.parse(webhook_endpoint.webhook_url)
+      request = Net::HTTP::Post.new(uri.request_uri)
+      request['Content-Type'] = 'application/json'
+      request.body = payload.to_json
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = (uri.scheme == 'https')
+      response = http.request(request)
+
+      WebhookDelivery.create!(
+        event: event_type,
+        payload: payload,
+        course: course,
+        webhook_url: webhook_endpoint.webhook_url,
+        sent_at: Time.now,
+        status: response.code,
+        response_headers: response.header,
+        response_body: response.body
+      )
+    end
```

Comment:
This seems unsafe. I'm not certain that we can always expect the `request` method call to always return a `response` object; it can raise exceptions (`Net::ReadTimeout`, for example), which would cause this job to fail without an entry ever being created in the `webhook_deliveries` table.

One possibility I can see is to append a `rescue` to this method and store the error class name:

```ruby
def perform(event_type, course, resource)
  # ...existing code
rescue => e
  WebhookDelivery.create!(
    error_class: e.class.name,
    # ...other available properties
  )
end
```